src/rgw/rgw_coroutine.cc

```diff
@@ -324,15 +324,15 @@ bool RGWCoroutinesStack::collect(RGWCoroutine *op, int *ret, RGWCoroutinesStack
       *ret = r;
       ldout(cct, 20) << "collect(): s=" << (void *)this << " stack=" << (void *)stack << " encountered error (r=" << r << "), skipping next stacks" << dendl;
       new_list.insert(new_list.end(), ++iter, s->entries.end());
-      done &= (iter != s->entries.end());
+      need_retry |= (iter != s->entries.end());
```

Comment:
'='  is enough and more clear, since next break will jump out of the loop

Reply:
Yes, you're right 👍. I have modified it.